### PR TITLE
Environment support

### PR DESCRIPTION
**Pull request by hansl**

### CODE REVIEW

File: packages/angular_devkit/build_webpack/test/browser/replacements_spec_large.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        import { meaning } from './meaning';
+
+        console.log(meaning);
+      `,
+    });
+
+    const overrides = {
+      fileReplacements: [
+        {
+          from: join(workspaceRoot, normalize('/src/meaning.ts')),
```
Comment (filipesilva) [inline packages/angular_devkit/build_webpack/test/browser/replacements_spec_large.ts]: This path should be relative to workspace? So just `normalize('/src/meaning.ts')`.

File: packages/angular_devkit/build_webpack/src/browser/schema.json [hunk 2]
```diff
@@ -59,9 +59,20 @@
       "description": "Defines the optimization level of the build.",
       "default": false
     },
-    "environment": {
-      "type": "string",
-      "description": "Defines the build environment."
+    "fileReplacements": {
+      "description": "Replace files with other files in the build.",
+      "type": "array",
+      "items": {
```
Comment (filipesilva) [inline packages/angular_devkit/build_webpack/src/browser/schema.json]: Can you add a default to `[]`?

Comment (hansl) [reply packages/angular_devkit/build_webpack/src/browser/schema.json]: Done

File: packages/angular_devkit/build_webpack/test/browser/replacements_spec_large.ts (hunk 1 above)
Comment (hansl) [reply packages/angular_devkit/build_webpack/test/browser/replacements_spec_large.ts]: Done

**Pull request merged**